### ISSUE TITLE
[Bug]: FROST KDA forward hangs with batch_invariant=True at T=16224

### ISSUE THREAD

**Issue by mdy666** [NVIDIA/cudnn-frontend#1052]
On B300, this forward call hangs with finite, all-zero inputs. It does not finish within 60 seconds. Changing only `T` from `16224` to `16384` completes successfully and returns exactly zero.

Reproduced with Python kernel sources from `develop` at `b2712de2da3832e9ad607867d22b2a1429c89cdc`, cuDNN Frontend native extension 1.29.0, backend 9.24.0, PyTorch 2.9.0 / CUDA 13.0, CuTe DSL 4.7.0, NVIDIA B300 (SM103).

Save as `repro.py` and run `timeout -k 5 60 python3 -u repro.py`:

```python
import torch
from cudnn.linear_attention import kimi_delta_attention

torch.cuda.set_device(0)
T, H, D = 16224, 128, 128
x = torch.zeros((T, H, D), device="cuda", dtype=torch.bfloat16)
beta = torch.ones((T, H), device="cuda", dtype=torch.bfloat16)
cu = torch.tensor([0, T], device="cuda", dtype=torch.int32)
a_log = torch.zeros(H, device="cuda")
dt_bias = torch.full((H, D), -5.5, device="cuda")
torch.cuda.synchronize()
print("Calling KDA", flush=True)
out = kimi_delta_attention(
    q=x, k=x, v=x, g=x, beta=beta, cu_seqlens=cu,
    a_log=a_log, dt_bias=dt_bias,
    safe_gate=True, gate_lower_bound=-5.0, batch_invariant=True,
    initial_state=None, output_final_state=False, plan_name="kda_frost",
)[0]
print("Forward returned; synchronizing", flush=True)
torch.cuda.synchronize()
assert torch.equal(out, torch.zeros_like(out))
print("PASS", flush=True)
```

Expected: the call completes with an all-zero output for either length.

Observed: `T=16224` prints `Calling KDA` and times out (exit 124); `T=16384` reaches `PASS` in a fresh process.

**Comment by mdy666** [comment]
The failure is narrowed down to the two chunk-factor warps in `kda_summary_f16.py::super_mma_warp`.

They can independently enter the next work item. After an odd-length item, restarting each warp at the same **local** parity changes its ring-slot ownership and can match a stale barrier phase. Keeping the **cumulative** chunk parity fixes the reproducer:

```diff
-        for local_chunk_idx in cutlass.range(chunk_parity, num_chunks_tile, 2, unroll=1):
+        first_chunk = chunk_parity ^ (cum_chunk_base % 2)
+        for local_chunk_idx in cutlass.range(first_chunk, num_chunks_tile, 2, unroll=1):
```

I checked the change in isolation: changing only the factor warps completes the failing case; changing only the CG0 parity does not. No CG0 change is needed.

Validation so far:

- The all-zero reproducer completes.
- Six small synthetic cases with odd/even lengths, ragged boundaries and optional state complete forward/backward and pass the existing 2% forward / 4% backward FP64-reference thresholds. These force one summary CTA to process multiple work items.
- A 32K-token ragged case that previously hung now completes forward/backward with finite results.
- On an unaffected 32K single-sequence case, output and all seven gradients have identical byte hashes before/after the change.

This change only affects scheduling; it does not change precision or the mathematical operations.

**Comment by mdy666** [comment]
Submitted the minimal fix as https://github.com/NVIDIA/cudnn-frontend/pull/1064.

The PR is based on current `develop` (`64f6b391`), with only the factor-warp loop-start change, matching descriptions, and a regression test. Revalidated that exact source on B300: both odd/even zero-reference tests pass, and all six nonzero FP64-reference cases pass. No numerical algorithm or precision change is included.

**Comment by jhjpark** [comment]
Thanks for reporting and fixing this!
